### PR TITLE
Add tags to a build instance

### PR DESCRIPTION
**Pull request by shubham149**
# Commit Checklist

Thank you for creating a pull request! To help us review / merge this can you make sure that your PR adheres as much as possible to the following.

## The Basics

If you are adding new functionality, please provide evidence of the new functionality.

### CODE REVIEW

File: internal/drivers/google/driver.go [hunk 1]
```diff
@@ -244,6 +244,26 @@ func (p *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (in
 	return &instanceMap, nil
 }
 
+func (p *config) SetTags(ctx context.Context, instanceID string, tags map[string]string) error {
+	logr := logger.FromContext(ctx).
```
Comment (vistaarjuneja) [inline internal/drivers/google/driver.go:248]: Can we return immediately if tags are empty? We don't need to do a get if there are no tags

File: internal/drivers/amazon/driver.go [hunk 2]
```diff
@@ -384,6 +384,21 @@ func (p *config) Logs(ctx context.Context, instanceID string) (string, error) {
 	return string(decoded), nil
 }
 
+func (p *config) SetTags(ctx context.Context, instanceID string,
+	tags map[string]string) error {
+	in := &ec2.CreateTagsInput{
```
Comment (vistaarjuneja) [inline internal/drivers/amazon/driver.go:389]: Likewise, if tags are empty we should return immediately

Comment (shubham149) [reply internal/drivers/amazon/driver.go:389]: Added the check in manager.go

File: internal/drivers/google/driver.go (hunk 1 above)
Comment (shubham149) [reply internal/drivers/google/driver.go:248]: Added the check in manager.go

**Review by vistaarjuneja — APPROVED** [review]

**Pull request merged**